src/VisualStudio/CSharp/Test/PersistentStorage/CloudCachePersistentStorageTests.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.SQLite.v2;
+using Microsoft.CodeAnalysis.Storage;
+using Microsoft.CodeAnalysis.UnitTests.WorkspaceServices.Mocks;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.ServiceHub.Framework.Services;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Cache;
+using Microsoft.VisualStudio.Cache.SQLite;
+using Microsoft.VisualStudio.RpcContracts.Caching;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests.WorkspaceServices
+{
+    public class CloudCachePersistentStorageTests : AbstractPersistentStorageTests
```

Comment:
all our storage tests are in the abstract base.  the individual provider tests just instantiate hte provider, which all tests run against.  this way we can do run all our heavily concurrent/abusive tests against all providers.